src/poetry/utils/shell.py

```diff
@@ -67,8 +68,20 @@ def get(cls) -> "Shell":
         return cls._shell
 
     def activate(self, env: "VirtualEnv") -> Optional[int]:
+        activate_script = self._get_activate_script()
+        bin_dir = "Scripts" if WINDOWS else "bin"
+        activate_path = env.path / bin_dir / activate_script
+
         if WINDOWS:
-            return env.execute(self.path)
+            if self._name == "powershell":
+                args = ["-NoExit", "-File", str(activate_path)]
+            else:
+                # /K will execute the bat file and
+                # keep the cmd process from terminating
+                args = ["/K", str(activate_path)]
+            proc = subprocess.Popen([self.path] + args)
+            proc.communicate()
```

Comment:
Since we're not actually using pipes to read from/write to stdout/stdin, `Popen.wait()` should be more appropriate: https://docs.python.org/3/library/subprocess.html#subprocess.Popen.wait

Reply:
Is there any reason not to use `subprocess.run()`?  The Python [doc](https://docs.python.org/3/library/subprocess.html#using-the-subprocess-module) says

> The recommended approach to invoking subprocesses is to use the run() function for all use cases it can handle

Reply:
That's a good point, I forgot that Python 3.4 has been out of support for a while. There is no reason not to use `run()` here as best I can tell.

Reply:
using subprocess.run() now